distribution/lib/Standard/Google_Api/0.0.0-dev/src/Google_Analytics_Account.enso

```diff
@@ -0,0 +1,108 @@
+
+import Standard.Table.Rows_To_Read.Rows_To_Read
+
+import project.Google_Analytics.Google_Analytics
+import project.Google_Credential.Google_Credential
+
+polyglot java import org.enso.google.GoogleAnalyticsReader.AnalyticsAccount
+
+## Type describing a Google Analytics account.
+type Google_Analytics_Account
+    private Value java_record
+
+    ## Account ID.
+    id : Text
+    id self = self.java_record.id
+
+    ## Account display name.
+    name : Text
+    name self = self.java_record.displayName
+
+    ## Account creation time.
+    create_time : Date_Time
+    create_time self = self.java_record.created
+
+    ## Account region code.
+    region_code : Text
+    region_code self = self.java_record.regionCode
+
+    ## Has the account been deleted?
+    is_deleted : Boolean
+    is_deleted self = self.java_record.deleted
+
+    ## PRIVATE
+    to_text : Text
+    to_text self = "Google_Analytics_Account " + self.name + " (" + self.id + ")"
+
+    ## PRIVATE
+    to_display_text : Text
+    to_display_text self = "GA Account {" + self.name + " (" + self.id + ")}"
+
+    ## ICON data_input
+       List of all properties of the account.
+
+       Arguments:
+       - credentials: The Google credentials to use. Default is the default
+         credentials.
+       - limit: The maximum number of accounts to read. Default is 1000.
+       - include_deleted: Whether to include deleted accounts. Default is false.
+    @credentials Google_Credential.default_widget
+    @limit Rows_To_Read.default_widget
+    properties : Google_Credential -> Rows_To_Read -> Boolean -> Vector
+    properties self credential:Google_Credential=..Default (limit : Rows_To_Read = ..First_With_Warning 1000) include_deleted:Boolean=False =
+        Google_Analytics.list_properties self credential limit include_deleted
+
+## Filter for Google Analytics accounts.
+type Google_Analytics_Account_Filter
+    All_Accounts
+
+    ## Filter to a single account
+    @account (Text_Input display=..Always)
+    Account account:Text|Google_Analytics_Account
+
+    ## Filter to set of accounts
+    @accounts (Vector_Editor item_editor=Text_Input item_default="''")
+    Accounts accounts:Vector
+
+    ## PRIVATE
+    default_widget : Any -> Any -> Widget
+    default_widget self_arg cache =
+        _ = self_arg
+        default = [Option "<All>" "..All_Accounts", Option "<Account>" "..Account", Option "<Multiple Accounts>" "..Accounts"]
+        Panic.catch Any handler=_->(make_single_choice default) <|
+            credentials = cache.if_not_nothing <| cache "credentials"
+            options = Google_Analytics.list_accounts credentials=(credentials.if_nothing Google_Credential.Default) limit=(..First 10) . map p->(Option (p.name + " (" + p.id + ")") "..Account "+p.id.pretty)
+            make_single_choice default+options display=..Always
+
+   ## PRIVATE
+   as_java : Array | Nothing
+   private as_java self = case self of
```

Comment:
nitpick, feel free to ignore: I think in new code maybe we could start favoring the checked type signatures, wherever it makes sense:
```suggestion
   private as_java self -> Array | Nothing = case self of
```

Especially as the 'old style' signature looks weird when combined with `private`.